packages/web-components/fast-ssr/src/template-parser/attributes.ts

```diff
@@ -0,0 +1,14 @@
+/**
+ * Extracts the attribute type from an attribute name
+ */
+export const attributeTypeRegExp = /([:?@])?(.*)/;
```

Comment:
Since the `html` tagged template function does this internally and associates it with the directive, do we also need it here? Is there some way the SSR parser could use what is already available?

Reply:
After this processing when we're rendering the element, we need to know which API to call `setProperty`, `setAttribute` or event delegation configuration. This regex essentially captures that information. Is that something we can infer from the directive type?

Reply:
I think we can make that information available. The binding will know that. It probably just needs to be officially exposed.